src/lib/import_export/binary/binary_writer.hpp

```diff
@@ -223,6 +224,37 @@ class BinaryWriter {
   template <typename T>
   static void _write_segment(const LZ4Segment<T>& lz4_segment, bool column_is_nullable, std::ofstream& ofstream);
 
+  /**
+   * FSSTSegments are dumped with the following layout:
+   *
+   * Description                 | Type                                | Size in bytes
+   * --------------------------------------------------------------------------------------------------------
+   * Encoding Type               | EncodingType                        | 1
+   * Attribute vector compr. ID. | CompressedVectorTypeID              | 1
+   * Compressed values size      | uint32_t                            | 4
+   * Compressed values           | vector<unsigned char>               | Compressed values size * 1
+   * Vector compress. bit width  | uint8_t                             | 1
+   * Offset values               | uint8_t                             | (Rows + 1) * (vector compr. bit width) / 8
+   *                                                                     rounded up to next multiple of word (8 byte)
+   * Reference offsets size      | uint32_t                            | 4
+   * Reference offsets           | vector<uint64_t>                    | 64
+   * NULL values size            | uint32_t                            | 4
+   * NULL values¹                | vector<bool> (BoolAsByteType)       | size * 1
+   * # elements in a bucket      | uint32_t                            | 4
+   * Decoder version id          | uint64_t                            | 8
+   * Decoder zero terminated flag| char                                | 1
+   * Decoder symbol lengths      | vector<char>                        | 255
```

Comment:
Yes, although it sounds a little bit odd. In fact, it's the FSST-library internal implementation. 

The symbol lengths are between 1 and 8, so a single `char` interpreted as an integer is large enough for symbol lengths. For the same reason, `uint64_t` is large enough to hold symbols with up to 8-byte large sequences. Our intuition is, that symbols are saved as integers (instead of `char` arrays) for performance reasons. 

See `fsst.h`:

https://github.com/cwida/fsst/blob/fffb613071cb44319c0d6b743a8d6eafc2ed2ad7/fsst.h#L79-L84